# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: widget/windows/RemoteBackbuffer.h
# Commit: d69f53c5968c
# Full Hash: d69f53c5968cf816bf07c5664b274863c155d19f
# Author: Chris Martin <cmartin@mozilla.com>
# Date: 2020-02-20 09:35:08
# Regressor Bug: 1604412
# File Overlap Count: 2
# Description:
#   Bug 1616000 - Fix broken push notifications r=handyman
#   
#   Bug 1604412 enabled "remote backbuffer" compositing when using non-accelerated
#   layers. However, due to my misunderstanding of how nWindow and compositing
#   handles layered windows, I broke the usecase where a child window is
# ==============================================================================

diff -r 115974724061 -r d69f53c5968c widget/windows/RemoteBackbuffer.h
--- a/widget/windows/RemoteBackbuffer.h	Wed Feb 19 23:03:12 2020 +0000
+++ b/widget/windows/RemoteBackbuffer.h	Wed Feb 19 22:20:33 2020 +0000
@@ -25,10 +25,13 @@
   Provider();
   ~Provider();
 
-  bool Initialize(HWND aWindowHandle, DWORD aTargetProcessId);
+  bool Initialize(HWND aWindowHandle, DWORD aTargetProcessId,
+                  nsTransparencyMode aTransparencyMode);
 
   Maybe<RemoteBackbufferHandles> CreateRemoteHandles();
 
+  void UpdateTransparencyMode(nsTransparencyMode aTransparencyMode);
+
   Provider(const Provider&) = delete;
   Provider(Provider&&) = delete;
   Provider& operator=(const Provider&) = delete;
@@ -49,6 +52,8 @@
   bool mStopServiceThread;
   std::thread mServiceThread;
   std::unique_ptr<PresentableSharedImage> mBackbuffer;
+  mozilla::Atomic<nsTransparencyMode, MemoryOrdering::Relaxed>
+      mTransparencyMode;
 };
 
 class Client {
